src/MapUtil/MapUtil.ts

```diff
@@ -475,6 +477,31 @@ export class MapUtil {
     });
   }
 
+  static calculateScaleAndCenterForExtent(olMap: OlMap, extent: OlExtent): {
+    center: OlCoordinate;
+    scale: number;
+  } | undefined {
+    if (_isNil(olMap) || _isNil(extent) || extent.length !== 4) {
+      return;
+    }
+    const view = olMap.getView();
+    const resolution = view.getResolutionForExtent(extent);
+    const unit = view.getProjection().getUnits() as Units;
+    const scale = MapUtil.getScaleForResolution(resolution, unit);
+    const center: OlCoordinate = [
+      (extent[0] + extent[2]) / 2,
+      (extent[1] + extent[3]) / 2
+    ];
```

Comment:
Non blocking, but we could also use [getCenter()](https://openlayers.org/en/latest/apidoc/module-ol_extent.html#.getCenter) here.

Reply:
Since the view is not necessarily fitted to the extent, I prefer the usage of manual center calculation here.